### DISCUSSION TITLE
Does it support training Mandarin-English Code-Switching?

### DISCUSSION THREAD

**Discussion by Minxiangliu** [NVIDIA-NeMo/NeMo#2708] [Q&A] [answered]
Hi,
I try to use nvidia nemo to train Mandarin-English ([like this](https://github.com/peter-yh-wu/code_mix)), It is possible to adjust the Code-Switching ASR?
if this is work, approximately how much I might need to prepare to train well?

Thanks for your time.

**Comment by titu1994** [comment, marked as answer]
We currently dont support code switching ASR, but it should be possible to train multilingual ASR. We haven't tried it out yet, so it's unknown how well it would perform though.

**Reply by Minxiangliu** [reply]
Thanks for your reply.
Do you mean that a model can recognize different languages, but can only output a single language?

**Reply by titu1994** [reply]
We haven't tried it, so I can't say for sure.

**Reply by Minxiangliu** [reply]
If I want to train Chinese, should the `text content` in the manifest be separated by `each word` or `a single font`?

**Reply by Slyne** [reply]
@Minxiangliu  text:"你今天吃KFC了嘛"  and the config file should be similar to : https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/quartznet/quartznet_15x5_zh.yaml

**Reply by Minxiangliu** [reply]
@Slyne  If I want ASR to be better, do I need to add more words to the label of the configuration file?

**Reply by Slyne** [reply]
@Minxiangliu The ```quartznet_15x5_zh.yaml``` is character/hanzi/alphabet based not word based.
